### PR TITLE
Expand numberOfPowershellRefAssemblies list capacity

### LINKED ISSUE #12815: Verify defaultRefAssemblies list capacity
static test to verfiy number of reference assemblies, see https://github.com/PowerShell/PowerShell/pull/12520#issuecomment-623543851

### PR DESCRIPTION
**Pull request by xtqqczze**
# PR Summary

* Increase the list capacity because .NET v5.0.100-preview.5.20278.13 has an extra assembly
* Remove assert added in #12520

## PR Context

HEAD of master has been broken since 99da109 (#12772), when .NET was updated to 5.0.100-preview.5.20278.13

https://github.com/PowerShell/PowerShell/issues/12815#issuecomment-636132717

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

**Comment by adityapatwardhan** [comment]
@xtqqczze Please use a more descriptive PR title.

**Comment by xtqqczze** [comment]
This resolves the issue in https://github.com/PowerShell/PowerShell/issues/12815#issuecomment-636132717 but does not fix #12815 as it does not include a static test.

**Comment by xtqqczze** [comment]
@TravisEz13 can you help merge to resolve the breakage described in https://github.com/PowerShell/PowerShell/issues/12815#issuecomment-636132717

**Comment by vexx32** [comment]
@TravisEz13 and @adityapatwardhan probably best to get this merged sooner than later; folks can't currently run debug builds at all while this is an issue.

**Comment by xtqqczze** [comment]
@iSazonov since you merged #12520, which this PR partially reverts, can you help merge this PR?

**Comment by xtqqczze** [comment]
The capacity is now 161 as per @iSazonov suggestion.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/AddType.cs [hunk 1]
```diff
@@ -685,11 +685,6 @@ private static IEnumerable<PortableExecutableReference> InitDefaultRefAssemblies
             // Add System.Management.Automation.dll
             defaultRefAssemblies.Add(MetadataReference.CreateFromFile(typeof(PSObject).Assembly.Location));
 
-            // We want to avoid reallocating the internal array, so we assert if the list capacity has increased.
-            Diagnostics.Assert(
-                defaultRefAssemblies.Capacity <= capacity,
-                $"defaultRefAssemblies was resized because of insufficient initial capacity! A capacity of {defaultRefAssemblies.Count} is required.");
-
```
[the remark this answers is not in the document]
Comment (xtqqczze) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/AddType.cs]: I'm not so sure a hard assert is desirable. We should remove it once we determine the const statically.

**Comment by xtqqczze** [comment]
@TravisEz13 ready to merge?

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/AddType.cs [hunk 2]
```diff
@@ -670,11 +670,9 @@ private void LoadAssemblies(IEnumerable<string> assemblies)
         /// </summary>
         private static IEnumerable<PortableExecutableReference> InitDefaultRefAssemblies()
         {
-            // Define number of reference assemblies distributed with PowerShell.
-            // This number is accurate as of PowerShell v7.1.0-preview.3 built with .NET v5.0.100-preview.5.20278.13
-            const int numberOfPowershellRefAssemblies = 152;
+            const int maxPowershellRefAssemblies = 160;
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/AddType.cs:673]:
```suggestion
            // Define number of reference assemblies distributed with PowerShell.
            const int maxPowershellRefAssemblies = 160;
```

**Review by iSazonov — APPROVED** [review]

**Pull request merged**